From 5c80837ea913179a6a983076c4adbab42e6a31dc Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 1 Mar 2017 11:37:59 +0100
Subject: git: update to v2.12.0

Update to git version v2.12.0: With commit 8aee769f (pathspec: copy and free
owned memory) member 'match' of 'struct pathspec_item' is no longer
'const char*' but just 'char*'.
---
 cgit.h      | 4 ++--
 shared.c    | 4 ++--
 ui-blob.c   | 2 +-
 ui-blob.h   | 2 +-
 ui-commit.c | 2 +-
 ui-commit.h | 2 +-
 ui-diff.c   | 4 ++--
 ui-diff.h   | 2 +-
 10 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/cgit.h b/cgit.h
index fbc6c6a..3eb04de 100644
--- cgit.h
+++ cgit.h
@@ -355,10 +355,10 @@ extern int cgit_diff_files(const struct object_id *old_oid,
 
 extern void cgit_diff_tree(const struct object_id *old_oid,
 			   const struct object_id *new_oid,
-			   filepair_fn fn, const char *prefix, int ignorews);
+			   filepair_fn fn, char *prefix, int ignorews);
 
 extern void cgit_diff_commit(struct commit *commit, filepair_fn fn,
-			     const char *prefix);
+			     char *prefix);
 
 __attribute__((format (printf,1,2)))
 extern char *fmt(const char *format,...);
diff --git a/shared.c b/shared.c
index c63f1e3..4602123 100644
--- shared.c
+++ shared.c
@@ -336,7 +336,7 @@ int cgit_diff_files(const struct object_id *old_oid,
 
 void cgit_diff_tree(const struct object_id *old_oid,
 		    const struct object_id *new_oid,
-		    filepair_fn fn, const char *prefix, int ignorews)
+		    filepair_fn fn, char *prefix, int ignorews)
 {
 	struct diff_options opt;
 	struct pathspec_item item;
@@ -367,7 +367,7 @@ void cgit_diff_tree(const struct object_id *old_oid,
 	diff_flush(&opt);
 }
 
-void cgit_diff_commit(struct commit *commit, filepair_fn fn, const char *prefix)
+void cgit_diff_commit(struct commit *commit, filepair_fn fn, char *prefix)
 {
 	const struct object_id *old_oid = NULL;
 
diff --git a/ui-blob.c b/ui-blob.c
index 5f30de7..c759520 100644
--- ui-blob.c
+++ ui-blob.c
@@ -33,7 +33,7 @@ static int walk_tree(const unsigned char *sha1, struct strbuf *base,
 	return 0;
 }
 
-int cgit_ref_path_exists(const char *path, const char *ref, int file_only)
+int cgit_ref_path_exists(char *path, const char *ref, int file_only)
 {
 	struct object_id oid;
 	unsigned long size;
diff --git a/ui-blob.h b/ui-blob.h
index 16847b2..d4be56a 100644
--- ui-blob.h
+++ ui-blob.h
@@ -1,7 +1,7 @@
 #ifndef UI_BLOB_H
 #define UI_BLOB_H
 
-extern int cgit_ref_path_exists(const char *path, const char *ref, int file_only);
+extern int cgit_ref_path_exists(char *path, const char *ref, int file_only);
 extern int cgit_print_file(char *path, const char *head, int file_only);
 extern void cgit_print_blob(const char *hex, char *path, const char *head, int file_only);
 
diff --git a/ui-commit.c b/ui-commit.c
index db69d54..47b9025 100644
--- ui-commit.c
+++ ui-commit.c
@@ -13,7 +13,7 @@
 #include "ui-diff.h"
 #include "ui-log.h"
 
-void cgit_print_commit(char *hex, const char *prefix)
+void cgit_print_commit(char *hex, char *prefix)
 {
 	struct commit *commit, *parent;
 	struct commitinfo *info, *parent_info;
diff --git a/ui-commit.h b/ui-commit.h
index 8198b4b..3d19eeb 100644
--- ui-commit.h
+++ ui-commit.h
@@ -1,6 +1,6 @@
 #ifndef UI_COMMIT_H
 #define UI_COMMIT_H
 
-extern void cgit_print_commit(char *hex, const char *prefix);
+extern void cgit_print_commit(char *hex, char *prefix);
 
 #endif /* UI_COMMIT_H */
diff --git a/ui-diff.c b/ui-diff.c
index 173d351..e8e9897 100644
--- ui-diff.c
+++ ui-diff.c
@@ -189,7 +189,7 @@ static void inspect_filepair(struct diff_filepair *pair)
 
 static void cgit_print_diffstat(const struct object_id *old_oid,
 				const struct object_id *new_oid,
-				const char *prefix)
+				char *prefix)
 {
 	int i;
 
@@ -382,7 +382,7 @@ void cgit_print_diff_ctrls(void)
 }
 
 void cgit_print_diff(const char *new_rev, const char *old_rev,
-		     const char *prefix, int show_ctrls, int raw)
+		     char *prefix, int show_ctrls, int raw)
 {
 	struct commit *commit, *commit2;
 	const unsigned char *old_tree_sha1, *new_tree_sha1;
diff --git a/ui-diff.h b/ui-diff.h
index 39264a1..4d5a83c 100644
--- ui-diff.h
+++ ui-diff.h
@@ -4,7 +4,7 @@
 extern void cgit_print_diff_ctrls(void);
 
 extern void cgit_print_diff(const char *new_hex, const char *old_hex,
-			    const char *prefix, int show_ctrls, int raw);
+			    char *prefix, int show_ctrls, int raw);
 
 extern struct diff_filespec *cgit_get_current_old_file(void);
 extern struct diff_filespec *cgit_get_current_new_file(void);
-- 
cgit v1.1-7-g6221

